src/compositor/input/pointer.rs

```diff
@@ -80,16 +88,16 @@ impl PointerHandler for Pointer {
 /// After the cursor has been warped, send pointer motion events to the view
 /// under the pointer or update the position of a view that might have been
 /// affected by an ongoing interactive move/resize operation
-fn update_view_position(cursor: &mut Cursor, seat: &mut Seat, views: &mut [View], time_msec: u32) {
+fn update_view_position(cursor: &mut Cursor,
+                        seat: &mut Seat,
+                        views: &mut Vec<Rc<View>>,
+                        time_msec: u32) {
     match seat.action {
         Some(Action::Moving { start }) => {
-            if let Some(focused_shell) = seat.focused.as_ref().map(|f| f.shell.clone()) {
-                for v in views {
-                    if v.shell == focused_shell {
-                        move_view(seat, cursor, v, start);
-                    }
-                }
-            }
+            seat.focused = seat.focused.take().map(|f| {
```

Comment:
I wish that worked because it's a common problem for me. I can't take a reference to `focused` without borrowing the seat. And I need to change the action of the seat so it needs to be mutable.

```
  --> src/compositor/input/pointer.rs:87:27
   |
86 |             if let Some(ref focused) = seat.focused {
   |                         ----------- immutable borrow occurs here
87 |                 move_view(seat, cursor, focused.clone(), start);
   |                           ^^^^ mutable borrow occurs here
88 |             }
   |             - immutable borrow ends here
```

Reply:
Ok :(